File: Algos2/LC_CS/299.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Solution299
{
    public string GetHint(string secret, string guess)
    {

        int cows = 0;
        int bulls = 0;
        Dictionary<char, int> count = new Dictionary<char, int>();

        for(int i=0; i < secret.Length; i++)
        {
            if (secret[i] == guess[i])
            {
                bulls += 1;

            }
            else
            {
                if (count.ContainsKey(secret[i]))
                {
                    count[secret[i]] += 1;
                }
                else
                {
                    count.Add(secret[i], 1);
                }

            }

        }

        for (int i = 0; i < secret.Length; i++)
        {
            if(guess[i] != secret[i] && count.ContainsKey(guess[i]) && count[guess[i]] > 0)
            {
                count[guess[i]] -= 1;
                cows += 1;
            }
        }


        return string.Format("{0}A{1}B", bulls, cows);

    }
}

File: Algos2/Algos2/LC/299.py
from collections import defaultdict

class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """

        bulls = 0
        cows = 0
        secret = list(secret)
        guess = list(guess)
        s = defaultdict(lambda : 0)
        g = defaultdict(lambda : 0)
        for i in range(len(secret)):
            if secret[i] == guess[i]:
                bulls += 1
                secret[i] = None
                guess[i] = None
                continue        

            g[guess[i]]+=1


            
        for i in range(len(secret)):
            if secret[i] == None:
                continue
            
            if g[secret[i]] > 0:
                cows += 1
                g[secret[i]] -= 1
        
            
        return '{}A{}B'.format(bulls, cows)
        
            
            